index.js

```diff
@@ -490,14 +490,37 @@ exports.init = function (sbot, config) {
 
     function next(accept) {
       getConfirm(invite_id, accept, function (err, confirm) {
-        if(!confirm)
-          connectFirst(invite, function (err, rpc) {
-            if(err) return cb(err)
-            rpc.peerInvites.confirm(accept, function (err, confirm) {
-              //TODO: store confirms for us in the state.
-              cb(err, confirm)
-            })
+        if (err) return cb(err)
+        if(confirm) {
+          console.error('ssb-peer-invites: Invite already confirmed')
+          return // mix: send something else back in cb for client side?
+        }
+
+        connectFirst(invite, function (err, rpc) {
+          if(err) return cb(err)
+          rpc.peerInvites.confirm(accept, function (err, confirm) {
+            //TODO: store confirms for us in the state.
+            if (err) return cb(err)
+
+            pull(
+              invite.pubs,
```

Comment:
I think a much simpler way to do this is just check that `invite.pubs` have each been passed to `gossip.add()` then let the gossip scheduler handle connecting to them.

Reply:
ok I've changed to gossip.add

I don't know what gossip.add calls back with, so hopefully this code works.
Also caught an obvious bug - needed `pull.values(invite.pubs)` !

Reply:
Resolved?